### PR TITLE
execute .env.leave file when leaving a directory

### PR DESCRIPTION
**Pull request by chunyang-wen**
When used with virtualenv, we can source .env file when entering a directory and activate an environment. But when leaving it, it will not be automatically deactivated. This implementation provides a solution, it will source .env.leave file when leaving working directory. But it also contains issues, such as when entering an environment, we cannot enter subdirectories, because it will be automatically deactivated( We are LEAVING current working directory), but we can also put additional .env file under those directories.

### CODE REVIEW

**Comment by Gnouc** [comment]
@chunyang-wen Can you change the format to match current file.

**Comment by chunyang-wen** [comment]
@Gnouc What do you mean? It has no conflicts.

**Comment by Gnouc** [comment]
@chunyang-wen Using tab instead of spaces.

**Comment by chunyang-wen** [comment]
@Gnouc As you wish.

**Comment by chunyang-wen** [comment]
Please help to decide whether this PR should be merged. If not, please close this PR. Thanks in advance.

File: activate.sh [hunk 1]
```diff
@@ -1,7 +1,14 @@
 AUTOENV_AUTH_FILE="${AUTOENV_AUTH_FILE:-$HOME/.autoenv_authorized}"
 AUTOENV_ENV_FILENAME="${AUTOENV_ENV_FILENAME:-.env}"
+AUTOENV_ENV_LEAVE_FILENAME="${AUTOENV_ENV_LEAVE_FILENAME:-.env.leave}"
+# AUTOENV_ENABLE_LEAVE
 
 autoenv_init() {
+
+	if [ ! -z $AUTOENV_ENABLE_LEAVE ]; then
```
Comment (Gnouc) [inline activate.sh]: - `! -z` should be `-n` for simple.
- `$AUTOENV_ENABLE_LEAVE` must be double quotes `"$AUTOENV_ENABLE_LEAVE"`. Variables should always be quoted in shell scripts, except you want `split+glob`.

**Pull request merged**